### ISSUE TITLE
New feature: Allow headings to be refactored (update anchors)

### ISSUE THREAD

**Issue by danhaywood** [asciidoctor/asciidoctor-intellij-plugin#421]
Caveat: this is probably pretty tough to do, but worth raising...

If a heading is created then it can be referenced using an anchor, eg:

    == Main API
    
    yada yada

And then  

    See the xref:#main-api[main API]

Suppose I change this heading, eg:

    == Synchronous API
    
    yada yada

Then, that anchor is invalid.

What I would like is for F6 (refactor > rename) on the heading text, and then have the plugin automatically update all anchors :

    See the xref:#synchronous-api[main API]

Even more awesome would be if the text within the brackets could change, at least if that text is directly equivalent to the old text of the heading):

    See the xref:#synchronous-api[synchronous API]

**Comment by ahus1** [comment]
Thanks for sharing this idea. 

Please be aware the automatically generated IDs by the plugin are kind of guessed: If two headings provide the same auto-generated ID, Asciidoctor will start to number them. The plugin does some guessing here, but can't be right all the time.

Renaming titles and text in xref's brackets would also be some guessing: in your example it would be renaming but keeping the case of the text in the xref's brackets (in your example "Main" vs. "main"). 

Due to this guessing this issue is not very high on the priority list for now. Please discuss if it shouldn't. I'm also looking forward suggestions for strategies to "do the right thing". 

I'd recommend to make all anchors for xrefs explicit in your docs. This would allow users to create stable bookmarks. For anchors and the related block IDs renaming using F6 already works. Also "find usages".

**Comment by danhaywood** [comment]
Yeah, I quite appreciate that this is a fuzzy requirement, and that it's
current best practice (though annoying) to put in explicit anchors.

As a couple of ideas for the two points you raise...

1. if there is a clash and suffixes, then the plugin could perhaps show
this as a conflict and allow the user to inspect and resolve manually.

2. for the title in the brackets, it would only attempt this if it could
match the existing text with respect to some heuristics.  I have in mind
the "preserve case" functionality that intellij supports for find/replace.

In terms of priorities, this isn't the most urgent, I'm sure, but it would
be pretty magical if it could at some stage in the future be supported.
